### ISSUE TITLE
Crash while using DS4Windows virtual controller

### ISSUE THREAD

**Issue by celerizer** [hikari-no-yume/touchHLE#330]
### Describe the feature or issue.

Steps to reproduce:
- Load DS4Windows with a PlayStation controller connected
- Load a game with multiple "button to touch" bindings, in my case Crash Kart
- Hold two mapped buttons at the same time at any point
- Program ends with failed assertation

```
thread 'main' panicked at src\frameworks\uikit\ui_touch.rs:155:13:
assertion `left == right` failed
  left: 2
 right: 1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Register state immediately after panic:
         R0: 0x375b8210  R1: 0x374f42b0  R2: 0x00000000  R3: 0x00000000
         R4: 0x00000001  R5: 0x374f7700  R6: 0xffffff44  R7: 0xffffff38
         R8: 0x00000000  R9: 0x00000000 R10: 0x00000000 R11: 0x00000000
        R12: 0x00000000  SP: 0xffffff2c  LR: 0x0007f76f  PC: 0x0007f8e4
Attempting to produce stack trace for main thread:
 0. 0x7f8e4 (PC)
 1. 0x7f76f (LR)
 2. 0x2a14
Next FP (0x0) is outside the stack.
```

I found this while using DS4Windows to create an XInput controller for my PS4 controller, so it should be two controllers sending identical inputs. Unfortunately don't have any other controller to test with, so I'm not sure if using two natural controllers with different inputs also causes the crash. Using Windows x64

### By submitting this issue, I certify that…

- [X] I understand that linking to, discussing or asking for pirated content is not permitted, and will lead to an immediate ban.
- [X] I understand that “Abandonware” and illegal copies of free games are also pirated content.

**Comment by hikari-no-yume** [comment]
This must be about this line in `ui_touch.rs`:

```rust
assert_eq!(host_object.phase, UITouchPhaseStationary);
```

@ciciplusplus I think it was you who added the `UITouchPhaseStationary` thing, maybe you can figure out what's up with this?

**Comment by ciciplusplus** [comment]
Hi,

Couldn't reproduce with a single Xbox 360 controller and "Crash Bandicoot Nitro Kart 3D" on touchHLE v0.2.2
Pressing multiple buttons at the same time I only get warnings about multi-touch being disabled (which is expected for this game).

**Comment by celerizer** [comment]
Right, the issue is in two controllers sending inputs at the same time. Since touchHLE supports the DualShock 4 controller, DS4Windows was sending an identical set of inputs acting as an Xinput controller. The issue seems to occur with two events firing at the exact same time from "different" controllers.

**Comment by hikari-no-yume** [comment]
While fixing the crash would be good, I would caution that you'll have other broken controller behaviour if it's fixed. touchHLE basically merges all recognised controllers into a single controller and won't handle duplicated inputs well. You will need to either disable this third-party tool or somehow tell SDL2 (used by touchHLE) to ignore it (there might be some environment variable?).
